dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/internal/classloader/JarResourceHandler.java

```diff
@@ -130,6 +134,35 @@ public Manifest getManifest() throws IOException {
         return jarFile.getManifest();
     }
 
+    @Override
+    public Set<String> getClassPackages() {
+        try {
+            ensureOpen();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return Collections.emptySet();
+        }
+
+        Enumeration<JarEntry> enumer = jarFile.entries();
```

Comment:
Similar to my previous comment for `DirectoryResourceHandler.getPackages()`, once `jarFile` is set and `getClassPackages()` is called for the first time, there is no need to recompute the `packageNames` Set on subsequent calls to `getClassPackages()`.

Reply:
This addressed by my previous comments to `DirectoryResourceHandler`.